### ISSUE TITLE
Optimize cmp & jge

### ISSUE THREAD

**Issue by krasin** [llvm-dcpu16/llvm-dcpu16#78]
llvm assembly has a pretty common pattern:

```
 %cond = icmp eq i32 %a, %b
 br i1 %cond, label %IfEqual, label %IfUnequal
IfEqual:
  ret i32 1
IfUnequal:
  ret i32 0
```

icmp translates to 5 instructions (see #37) and jge translates into 2 instructions (see #36). At the same time, it can be translated into two instructions instead of 7:

```
IFE $reg1, $reg2
SET PC, label
```

**Comment by krasin** [comment]
ARM has instructions with conditional prefixes, may be we can steal related logic from there.

**Comment by Blei** [comment]
The start of an implementation for more correct BR_CC lowering can be found here: https://github.com/Blei/llvm-dcpu16/tree/optimize-brcc-v2

It seems to produce correct code in the examples I looked at, but there is still work to be done... And I'm not sure at all if the current implementation is correct or if i made some obvious mistakes. LLVM is still very huge and confusing to me.

**Comment by ghost** [comment]
For me it looks like you have at least a good understanding in what you are doing! Keep up the work!

**Comment by krasin** [comment]
+1, looking promising!

**Comment by Blei** [comment]
This has been fixed by #109.
